sourcemash/api/categories.py

```diff
@@ -1,11 +1,23 @@
 from . import api, login_required
-from flask.ext.restful import Resource, marshal, fields
+from sourcemash.database import db
+from flask.ext.restful import Resource, marshal, fields, reqparse, inputs
 from flask.ext.security import current_user
 
-from sourcemash.models import Item, Category, UserItem
+from sourcemash.models import Item, Category, UserItem, UserCategory
 
 from sqlalchemy import func
+from sqlalchemy.orm.exc import MultipleResultsFound, NoResultFound
 
+MIN_ITEM_COUNT = 1
+
+class isUnread(fields.Raw):
+    def output(self, key, category):
+        try:
+            unread = UserCategory.query.filter_by(user=current_user, category_id=category.id).one().unread
+        except (MultipleResultsFound, NoResultFound):
```

Comment:
No, it shouldn't. But if there is a db problem and it happens, we should catch it and return the "default" behavior as opposed to dumping a stacktrace.